tests/robustness/snapmeta/simple_test.go

```diff
@@ -1,25 +1,174 @@
+		todo         bool
+		key          string
+		indexUpdates map[string]IndexOperation
+	}
+
+	type testcase struct {
+		storeOperations  storeOperation
+		deleteOperations deleteOperation
+	}
+
+	testcases := []testcase{
+		{
+			storeOperations: storeOperation{
+				todo:         true,
+				key:          storeKey,
+				val:          []byte("some stored data"),
+				indexUpdates: nil,
+			},
+			deleteOperations: deleteOperation{
+				todo: false,
+			},
+		},
+		{
+			storeOperations: storeOperation{
+				todo: true,
+				key:  storeKey,
+				val:  []byte("some stored data"),
+				indexUpdates: map[string]IndexOperation{
+					idxName: AddToIndexOperation,
+				},
+			},
+			deleteOperations: deleteOperation{
+				todo: false,
+			},
+		},
+		{
+			storeOperations: storeOperation{
+				todo: true,
+				key:  storeKey,
+				val:  []byte("some stored data"),
+				indexUpdates: map[string]IndexOperation{
+					idxName: AddToIndexOperation,
+				},
+			},
+			deleteOperations: deleteOperation{
+				todo: true,
+				key:  storeKey,
+				indexUpdates: map[string]IndexOperation{
+					idxName: RemoveFromIndexOperation,
+				},
+			},
+		},
 	}
 
-	if got, want := idxKeys[0], storeKey; got != want {
-		t.Fatalf("expected key %v but got %v", want, got)
+	for _, v := range testcases {
+		simple := NewSimple()
+
+		if v.storeOperations.todo == true {
+			simple.Store(v.storeOperations.key, v.storeOperations.val, v.storeOperations.indexUpdates)
+		}
+
+		// Check if the value is loaded correctly
+		checkStoredData, err := simple.Load(v.storeOperations.key)
+		if err != nil {
+			t.Fatalf("expected %v keys but got %v", nil, err)
+		}
+
+		if !reflect.DeepEqual(checkStoredData, data) {
+			t.Fatalf("expected datas %v but got %v", data, checkStoredData)
+		}
+
+		if v.storeOperations.indexUpdates != nil {
+			for idx := range v.storeOperations.indexUpdates {
+				idxKeys := simple.GetKeys(idx)
+				if got, want := len(idxKeys), 1; got != want {
+					t.Fatalf("expected %v keys but got %v", want, got)
+				}
+
+				if got, want := idxKeys[0], v.storeOperations.key; got != want {
+					t.Fatalf("expected key %v but got %v", want, got)
+				}
+			}
+		}
+
+		if v.deleteOperations.todo == true {
+			simple.Delete(v.storeOperations.key, v.storeOperations.indexUpdates)
+		}
+
+		// Check if the value is deleted correctly
+		_, err = simple.Load(v.deleteOperations.key)
+		if err == nil {
+			t.Fatalf("expected %v keys but got %v", nil, err)
+		}
+
+		if !reflect.DeepEqual(checkStoredData, data) {
+			t.Fatalf("expected datas %v but got %v", data, checkStoredData)
+		}
+
+		if v.deleteOperations.indexUpdates != nil {
+			for idx := range v.deleteOperations.indexUpdates {
+				idxKeys := simple.GetKeys(idx)
+				if got, want := len(idxKeys), 1; got != want {
+					t.Fatalf("expected %v keys but got %v", want, got)
+				}
+
+				if got, want := idxKeys[0], v.deleteOperations.key; got != want {
+					t.Fatalf("expected key %v but got %v", want, got)
+				}
+			}
+		}
```

Comment:
I would have set it up something like this (for example):
```

func TestSimpleWithIndex(t *testing.T) {
	type opType int
	const (
		storeOpType opType = iota
		deleteOpType
	)

	type storeOp struct {
		op    opType
		key   string
		value []byte
		idxOp map[string]IndexOperation
	}

	type expResult struct {
		expErr   bool
		expValue []byte
	}

	for _, tc := range []struct {
		name            string
		storeOperations []storeOp
		expectedKVs     map[string]expResult // List of expected key-values
		expectedIdxKeys map[string][]string  // List of expected keys in the given index
	}{
		{
			name: "Test key value change",
			storeOperations: []storeOp{
				{
					op:    storeOpType,
					key:   "some-key",
					value: []byte("some-value"),
					idxOp: nil,
				},
				{
					op:    storeOpType,
					key:   "some-key",
					value: []byte("some-different-value"),
					idxOp: nil,
				},
			},
			expectedKVs: map[string]expResult{
				"some-key": expResult{
					expErr:   false,
					expValue: []byte("some-different-value"),
				},
			},
		},
	} {
		t.log(tc.name)

		simple := NewSimple()

		for _, op := range tc.storeOperations {
			switch op.op {
			case storeOpType:
				simple.Store(op.key, op.value, op.idxOp)
			case deleteOpType:
				simple.Delete(op.key, op.idxOp)
			}
		}

		// Check keys present and expected value
		for wantK, expResult := range tc.expectedKVs {
			gotV, err := simple.Load(wantK)
			if expResult.expErr != (err != nil) {
				// Failure
			}
			if bytes.Compare(expResult.expValue, gotV) != 0 {
				// Failure
			}
		}

		// Check indices contain expected keys
		for idxName, wantKeys := range tc.expectedIdxKeys {
			gotKeys := simple.GetKeys(idxName)
			if !compareSlices(gotKeys, wantKeys) {
				// Failure
			}
		}
	}
}

```
Keep the actual test execution as simple and generalized as possible. It essentially boils down to:
- iterate through a list of store operations to put the store in an expected state
- check the keys you expect to be there/not be there
- check the index contains the keys you expect

In the one testcase example I gave, you can check a key being first set to one value, then another.